var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -163,28 +168,38 @@ def determine_b2_options(self, spec, options):
             threadingOpts.append('single')
         if not threadingOpts:
             raise RuntimeError("At least one of {singlethreaded, multithreaded} must be enabled")
-        
+
         options.extend([
             'toolset=%s' % self.determine_toolset(spec),
             'link=%s' % ','.join(linkTypes),
             '--layout=tagged'])
-        
+
         return threadingOpts
 
     def install(self, spec, prefix):
+        # On Darwin, Boost expects the Darwin libtool. However, one of the
+        # dependencies may have pulled in Spack's GNU libtool, and these two are
+        # not compatible. We thus create a symlink to Darwin's libtool and add
+        # it at the beginning of PATH.
+        if sys.platform == 'darwin':
```

Comment:
I think the Mac-specific packages.yaml file is probably the right way to go.

BTW... it should be possible to have null paths for packages.  We don't
need to add "/usr", etc. to the path a zillion times.  Moreover, on Linux,
some things are in /usr/lib and some in /usr/lib64.  So when you say it's
in "/usr", that's not even really accurate.  So I would suggest a
packages.yaml file like:

```
packages:

  all:
    compiler: [clang, gcc]
  libtool:
    paths:
      libtool@2.4.6%clang@7.0.2-apple: <system>
    buildable: False # never build libtool

```


On Fri, Mar 18, 2016 at 8:44 AM, Denis Davydov <notifications@github.com>
wrote:

> In var/spack/repos/builtin/packages/boost/package.py:
>
> >          return threadingOpts
> >
> >      def install(self, spec, prefix):
> > +        # On Darwin, Boost expects the Darwin libtool. However, one of the
> > +        # dependencies may have pulled in Spack's GNU libtool, and these two are
> > +        # not compatible. We thus create a symlink to Darwin's libtool and add
> > +        # it at the beginning of PATH.
> > +        if sys.platform == 'darwin':
>
> i guess the same issue may appear in other places? Would that be better to
> force OS-X users somehow do something like
>
> packages:
>   all:
>     compiler: [clang, gcc]
>   libtool:
>     paths:
>       libtool@2.4.6%clang@7.0.2-apple: /usr/ # libtool is in /usr/bin/
>     buildable: False # never build libtool
>
> in packages.yaml? Maybe not only libtool, but some reasonable defaults
> for things which ship with OS-X XCode, i.e. zlib, etc?
>
> I think it is also related to this issue #541
> <https://github.com/LLNL/spack/issues/541>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/LLNL/spack/pull/563/files/05ed139e16a16c2b599ae5cde2176d706034b658#r56650056>
>